examples/stats/plot_sensor_regression.py

```diff
@@ -10,68 +10,38 @@
 and timepoint. Example also shows the T statistics and the associated
 p-values.
 
-Note that this example is for educational purposes and that the data used
-here do not contain any significant effect.
-
-(See Hauk et al. (2006). The time course of visual word recognition as
-revealed by linear regression analysis of ERP data. Neuroimage.)
+Here, we repeat a few of the analyses from Dufau et al. (2014) by accessing
+the metadata object, which contains word-level information about various
+psycholinguistically relevant features of the words for which we have EEG
+activity.
+
+For the general methodology, see e.g. Hauk et al. (2006). The time course of 
+visual word recognition as revealed by linear regression analysis of ERP data. 
+Neuroimage.)
 """
 # Authors: Tal Linzen <linzen@nyu.edu>
 #          Denis A. Engemann <denis.engemann@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.com>
 #
 # License: BSD (3-clause)
 
-import numpy as np
-
+import pandas as pd
 import mne
-from mne.datasets import sample
-from mne.stats.regression import linear_regression
-
-print(__doc__)
-
-data_path = sample.data_path()
-
-###############################################################################
-# Set parameters and read data
-raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
-event_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw-eve.fif'
-tmin, tmax = -0.2, 0.5
-event_id = dict(aud_l=1, aud_r=2)
-
-# Setup for reading the raw data
-raw = mne.io.read_raw_fif(raw_fname)
-events = mne.read_events(event_fname)
-
-picks = mne.pick_types(raw.info, meg='mag', eeg=False, stim=False,
-                       eog=False, exclude='bads')
-
-# Reject some epochs based on amplitude
-reject = dict(mag=5e-12)
-epochs = mne.Epochs(raw, events, event_id, tmin, tmax, proj=True,
-                    picks=picks, baseline=(None, 0), preload=True,
-                    reject=reject)
-
-###############################################################################
-# Run regression
-
-names = ['intercept', 'trial-count']
-
-intercept = np.ones((len(epochs),), dtype=np.float)
-design_matrix = np.column_stack([intercept,  # intercept
-                                 np.linspace(0, 1, len(intercept))])
-
-# also accepts source estimates
-lm = linear_regression(epochs, design_matrix, names)
-
+from mne.stats import linear_regression
+from mne.datasets import kiloword
 
-def plot_topomap(x, units):
-    x.plot_topomap(ch_type='mag', scalings=1., size=1.5, vmax=np.max,
-                   units=units, times=np.linspace(0.1, 0.2, 5))
+# Load the data
+path = kiloword.data_path() + '/kword_metadata-epo.fif'
+epochs = mne.read_epochs(path)
+print(epochs.metadata.head())
 
+# Add intercept column
+df = pd.DataFrame(epochs.metadata)
+epochs.metadata = df.assign(Intercept=[1 for _ in epochs.events])
```

Comment:
I would just do:

epochs.metadata['Intercept'] = 1.

Reply:
I know, I would too, if it worked :)

@choldgraf this does not work, I assume metadata is immutable or something?